### ISSUE TITLE
Linting workflow doing extra work

### ISSUE THREAD

**Issue by Molkree** [ooipy/ooipy#84]
Because you utilize pre-commit.ci now, which runs on PRs, your [linting workflow](https://github.com/ooipy/ooipy/blob/1ef908b221f58b1a016878589f0474ed494d7315/.github/workflows/linting.yml) does double work of checking all PRs
https://github.com/ooipy/ooipy/blob/1ef908b221f58b1a016878589f0474ed494d7315/.github/workflows/linting.yml#L4

In addition to that, pre-commit Action you use was [deprecated](https://github.com/pre-commit/action#:~:text=README.md-,DEPRECATED,-this%20action%20is) recently.
https://github.com/ooipy/ooipy/blob/1ef908b221f58b1a016878589f0474ed494d7315/.github/workflows/linting.yml#L12
![image](https://user-images.githubusercontent.com/12860284/127576331-e30fad30-8d4e-4d95-bbe5-5ef9a34586da.png)
It works for now but I'd probably just move to using pre-commit directly (without Action). This will also allow to have "granular" checks (one for Black, one for flake8, etc) if you so desire.

To sum up what I suggest:
1. Remove `pull_request` trigger (this is a no-brainer).
2. Maybe switch from Action to installing+calling pre-commit manually.
 2a. Maybe split one "linting" check into separate checks for each hook (as separate jobs in the workflow).

**Comment by lsetiawan** [comment]
@Molkree I think we should just remove this action all together if we are using the `pre-commit.ci`. Even they recommend that:

```
DEPRECATED this action is in maintenance-only mode and will not be accepting new features.

Please switch to using pre-commit.ci which is faster and has more features.
```

**Comment by Molkree** [comment]
> I think we should just remove this action all together

Yes, action should go away but workflow should stay because pre-commit.ci only lints files changed in the PR but ideally we'd want to run linting on all files from time to time (for example, linting tools evolve too).

> `pre-commit run --all-files`: run all the hooks against all the files. This is a useful invocation if you are using pre-commit in CI.

From pre-commit [docs](https://pre-commit.com/#:~:text=This%20is%20a%20useful%20invocation%20if%20you%20are%20using%20pre-commit%20in%20CI.).

I will send a PR.
